src/bootman/bootman.h

```diff
@@ -350,9 +352,35 @@ static inline void kernel_array_free(void *v)
         nc_array_free(&a, (array_free_func)free_kernel);
 }
 
+/**
+ * Get freestanding initrds (initrds without kernel modules inside)
+ */
+bool boot_manager_get_initrd_freestanding(BootManager *self);
+
+/**
+ * Copy freestanding initrd
+ */
+bool boot_manager_copy_initrd_freestanding(BootManager *self);
+
+/**
+ * Remove old freestanding initrd
+ */
+bool boot_manager_remove_initrd_freestanding(BootManager * self);
+
+/*
+ * Iterate initrd elements
+ */
+void boot_manager_initrd_iterator_init(const BootManager *manager, NcHashmapIter *iter);
+
+/**
+ * Get freestanding initrd to bootloader entry
+ */
+bool boot_manager_enumerate_initrd_freestanding(NcHashmapIter *iter, const char *name);
```

Comment:
You renamed a different function than I commented on. With the current function names, the suggestion is:

`boot_manager_get_initrd_freestanding` => `boot_manager_enumerate_initrds_freestanding` (this is your "init" function)

`boot_manager_initrd_iterator_init` => `boot_manager_initrd_iterator_init` (keep the name)

`boot_manager_enumerate_initrd_freestanding` => `boot_manager_initrd_iterator_next`

Reply:
Ok, name matter